File: 0392-is-subsequence/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i1=s.length();
        int i2=t.length();
        while(i1>0&&i2>0){
           if(s[i1-1]==t[i2-1]){
            i1--;
           }
           i2--;
        }
        if(i1==0){
            return true;
        }
        return false;
    }
};

File: 0392-is-subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if (len(s)==0):
            return True
        i=0
        n=len(s)
        for j in range(len(t)):
            if(s[i]==t[j]):
                i+=1
                if(i==n):
                    return True
        return False            
                
